seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,48 +1,112 @@
 """APIs for management of projects related to AnVIL workspaces."""
 
 import logging
-from django.contrib.auth.decorators import login_required
+import json
 
+from django.shortcuts import redirect
+from django.contrib.sites.shortcuts import get_current_site
+
+from seqr.models import Project, CAN_EDIT
+from seqr.views.utils.json_to_orm_utils import create_model_from_json
 from seqr.views.utils.json_utils import create_json_response
-from settings import API_LOGIN_REQUIRED_URL
+from seqr.views.utils.file_utils import load_uploaded_file
+from seqr.views.utils.terra_api_utils import add_service_account
+from seqr.views.utils.pedigree_info_utils import parse_pedigree_table
+from seqr.views.apis.individual_api import add_or_update_individuals_and_families
+from seqr.utils.communication_utils import send_load_data_email
+from seqr.views.utils.permissions_utils import google_auth_required, check_workspace_perm
 
 logger = logging.getLogger(__name__)
 
 
-@login_required(login_url=API_LOGIN_REQUIRED_URL)
+def add_individuals_and_families(project, individual_records, user):
+    _, updated_individuals = add_or_update_individuals_and_families(
```

Comment:
2 line helper methods that are called exactly once are never a good idea. Helper methods are useful if a) you need to do the same operation multiple times so you call the helper function multiple times or b) the functionality is very complicated so the main function gets too hard to read and the helper function makes the code clearer. But in this case it would be more readable if you just put these 2 lines of code directly in the main function